tests/opentelemetry-test-utils/src/opentelemetry/test/test_base.py

```diff
@@ -130,6 +137,93 @@ def disable_logging(highest_level=logging.CRITICAL):
         finally:
             logging.disable(logging.NOTSET)
 
+    def get_sorted_metrics(self):
+        resource_metrics = (
+            self.memory_metrics_reader.get_metrics_data().resource_metrics
+        )
+
+        all_metrics = []
+        for metrics in resource_metrics:
+            for scope_metrics in metrics.scope_metrics:
+                all_metrics.extend(scope_metrics.metrics)
+
+        return self.sorted_metrics(all_metrics)
+
+    @staticmethod
+    def sorted_metrics(metrics):
+        """
+        Sorts metrics by metric name.
+        """
+        return sorted(
+            metrics,
+            key=lambda m: m.name,
+        )
+
+    def assert_metric_expected(
+        self,
+        metric: Metric,
+        expected_data_points: Sequence[DataPointT],
+        est_value_delta: Optional[float] = 0,
+    ):
+        self.assertEqual(
+            len(expected_data_points), len(metric.data.data_points)
+        )
+        for expected_data_point in expected_data_points:
+            self.assert_data_point_expected(
+                expected_data_point, metric.data.data_points, est_value_delta
+            )
+
+    # pylint: disable=unidiomatic-typecheck
+    @staticmethod
+    def is_data_points_equal(
+        expected_data_point: DataPointT,
+        data_point: DataPointT,
+        est_value_delta: Optional[float] = 0,
+    ):
+        if type(expected_data_point) != type(data_point) or not isinstance(
+            expected_data_point, (HistogramDataPoint, NumberDataPoint)
+        ):
+            return False
+
+        values_diff = None
+        if isinstance(data_point, HistogramDataPoint):
+            values_diff = abs(expected_data_point.sum - data_point.sum)
```

Comment:
This should also be comparison of `count`, `min` and `max` but without delta diff.

Reply:
The comparison of max and min adds unnecessary complexity in my opinion.
For the instrument tests what matters is what the instrument specifically adds.
Like value, attributes, and counts.
The max and min are calculated from it and should be tested just in the metrics code and not for each instrumentation.
1. The tests in each instrumentation need to calculate the max and min of each expected data point, to compare to the data point that was created.
2. If the comparison is with delta diff the test needs to ignore the max and the min anyway